compiler/ballerina-compiler-api/src/main/java/org/ballerina/compiler/impl/BallerinaSemanticModel.java

```diff
@@ -99,4 +129,31 @@ public BallerinaSemanticModel(BLangCompilationUnit compilationUnit, BLangPackage
     public List<Diagnostic> diagnostics(TextRange range) {
         return new ArrayList<>();
     }
+
+    private boolean isSymbolInUserProject(BSymbol symbol, DiagnosticPos cursorPos) {
+        return symbol.origin == SOURCE &&
+                (cursorPos.compareTo(symbol.pos) > SYMBOL_POSITION ||
+                        symbol.owner instanceof BPackageSymbol ||
+                        Symbols.isFlagOn(symbol.flags, Flags.WORKER));
+    }
+
+    private boolean isImportedSymbol(BSymbol symbol) {
+        return symbol.origin == COMPILED_SOURCE &&
+                (Symbols.isFlagOn(symbol.flags, Flags.PUBLIC) || symbol.getKind() == SymbolKind.PACKAGE);
+    }
+
+    private boolean isWithinSymbol(LinePosition cursorPos, DiagnosticPos symbolPos) {
+        int symbolStartLine = symbolPos.getStartLine();
+        int symbolEndLine = symbolPos.getEndLine();
+        int symbolStartCol = symbolPos.getStartColumn();
+        int symbolEndCol = symbolPos.getEndColumn();
+        int cursorLine = cursorPos.line();
+        int cursorCol = cursorPos.offset();
+
+        return (symbolStartLine < cursorLine && symbolEndLine > cursorLine)
```

Comment:
Is it possible to get this logic to a common place? Since the same is implemented in https://github.com/ballerina-platform/ballerina-lang/pull/25605/files#diff-3bb78fa9a4bfafbf27ec61bfdea79e93R1059 as well?

Reply:
For now, I just simply took the same logic. But yeah, need to figure out where to place things like these